src/compiler/checker.ts

```diff
@@ -3953,7 +4264,7 @@ namespace ts {
             // bubble up to the declaration
             let currentNode: Node = typeReferenceNode;
             // forEach === exists
-            while (!forEach(typeParameterSymbol.declarations, d => d.parent === currentNode.parent)) {
+            while (!forEach(typeParameterSymbol.declarations, d => getTypeParameterOwner(d) === currentNode.parent)) {

```

Comment:
It seems like we're computing the same information in `getTypeParameterOwner` for each iteration. Doesn't look very efficient to me.